### ISSUE TITLE
Question about heatmap

### ISSUE THREAD

**Issue by Lrbomchz** [Haochen-Wang409/HPM#3]
Hello, it is very nice to read this paper. I am conducting research on 'learn where to mask' just like you. Could you please explain how the heatmap in your paper is generated and how to convert weights to RGB for the heatmap? I have tried several approaches, but none of them look as visually appealing as yours. Thank you very much.

**Comment by Haochen-Wang409** [comment]
Thanks a lot for your attention!
Here is the pseudo-code for generating an RGB heatmap using the predicted reconstruction loss:

```
def show_loss_pred(model, inputs, img, num_patches=196, h=14, w=14, input_size=224):
    # generate a full visible mask
    mask = np.zeros((num_patches)).astype(bool)
    mask = torch.from_numpy(mask).unsqueeze(0).bool()

    with torch.no_grad():
        outs = model(inputs, mask)

    loss_show = outs['loss_pred'][0].cpu()

    # postprocessing for predicted loss
    loss_show = loss_show - loss_show.mean()
    loss_show = np.exp(loss_show) / (1 + np.exp(loss_show))
    loss_show = loss_show / loss_show.max()
    loss_show = (loss_show * 255).numpy().astype(np.uint8)

    reconstruct = loss_show.copy()

    reconstruct = reconstruct.reshape((h, w)).astype(np.uint8)
    reconstruct_color = cv2.applyColorMap(reconstruct, cv2.COLORMAP_JET)
    reconstruct_color = np.array(cv2.cvtColor(reconstruct_color, cv2.COLOR_BGR2RGB))

    rec_big = np.zeros((input_size, input_size, 3)).astype(np.uint8)
    for i in range(h):
        for j in range(w):
            rec_big[i * 16: i * 16 + 16, j * 16: j * 16 + 16, :] = reconstruct_color[i, j]

    img = np.array(img)
    img_show = (0.5 * img + 0.5 * rec_big).astype(np.uint8)

    return img_show
```

**Comment by Lrbomchz** [comment]
Thank you very much, it‘s work!

**Comment by Haochen-Wang409** [comment]
Happy to hear that :)
I hope our work brings inspiration to you.
